loopin.go

```diff
@@ -835,44 +831,117 @@ func (s *loopInSwap) waitForSwapComplete(ctx context.Context,
 		return err
 	}
 
-	htlcSpend := false
+	// We synchronize access to the sweepFee once the timeout transaction
+	// has been published. If we already published the timeout transaction
+	// we don't have to wait for the sweepFee to be set.
+	sweepFeeWg := sync.WaitGroup{}
+	sweepFeeWg.Add(1)
+	if sweepFee > 0 {
+		// We published the timeout transaction.
+		sweepFeeWg.Done()
+	}
+
+	// We monitor the htlc spend by the server or the timeout sweep in a
+	// separate go routine since we don't want to block on them in order to
+	// continue with successive loop ins once the invoice is settled.
+	executorWg.Add(1)
+	go func(s *loopInSwap) {
+		defer executorWg.Done()
+		spent := false
+		notifier := s.lnd.ChainNotifier
+		for !spent {
```

Comment:
nit: I'd consider extracting the whole goroutine to a method to make it more readable.